app/models/case_management/icasework.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module CaseManagement
+  # Implementation of a CaseManagement for the iCasework Case Management System.
+  class Icasework
+    def initialize(client: nil)
+      @client = client || ::Icasework::Case
+    end
+
+    def submit_foi_request!(name:, email:, body:)
+      request = client.create(
+        request_method: 'Online Form',
+        type: 'InformationRequest',
+        name: name,
+        email: email,
+        details: body
```

Comment:
The raw post data should be:

```
     -d '{"Type":"InformationRequest",
           "RequestMethod":"Online Form",
           "Customer.Name":"Alice Brown",
           "Customer.Email":"alice@example.com",
           "Details":"test foi"}'
```

Do we convert these keys correctly. Looks like there [might be some todos](https://github.com/mysociety/icasework-ruby/blob/48a666bcae39ef3f1825027c7b4cd940450ad890/lib/icasework/resource.rb#L123-L126) for this to work as is?. I'd be inclined to use string keys so that we don't have to jump through hoops to translate between the "official" API and our client library. The point of this `CaseManagement` interface is that we only have to deal with the grossness in one place?

```ruby
client.create(
  'RequestMethod' => 'Online Form',
  'Type' => 'InformationRequest',
  'Customer.Name' => name,
  'Customer.Email' => email,
  'Details' => body
)
```

Reply:
From my tests the endpoint does accept `Name` and `Email`. Not saying to should uses these as is seems undocumented. My intention and some of the work in progress I've done on the gem would handle this nicely EG:
```ruby
client.create(
  request_method: 'Online Form',
  type: 'InformationRequest',
  customer: {
    name: name,
    email: email,
  },
  details: body
)
```

Reply:
> From my tests the endpoint does accept `Name` and `Email`

Worth posting a BC question about this and whether there's any difference.